samples/algorithms/sudoku-grover/ColoringGroverWithConstraints.qs

```diff
@@ -0,0 +1,365 @@
+                        } apply{
+                            // AND color's most significant bit with OR of least significant bits. 
+                            // This will set conflictQubit to 1 if color > 8
+                            CCNOT(color[3], tempQubit, conflictQubit);
+                        }
+                    }
+                }
+            } apply {
+                // If there are no conflicts (all qubits are in 0 state), the vertex coloring is valid
+                (ControlledOnInt(0, X))(conflictQubits, target);
+            }
+        }
+    }
+
+    /// # Summary
+    /// OR oracle for an arbitrary number of qubits in query register
+    operation Oracle_Or (queryRegister : Qubit[], target : Qubit) : Unit is Adj {        
+        // x₀ ∨ x₁ = ¬ (¬x₀ ∧ ¬x₁)
+        // First, flip target if both qubits are in |0⟩ state
+        (ControlledOnInt(0, X))(queryRegister, target);
+        // Then flip target again to get negation
+        X(target);
+    }
+    
+    /// # Summary
+    /// Using Grover's search to find vertex coloring.
+    ///
+    /// # Input
+    /// ## V
+    /// The number of Vertices in the graph
+    /// ## K
+    /// The number of bits per color
+    /// ## maxIterations
+    /// An estimate of the maximum iterations needed
+    /// ## oracle
+    /// The Oracle used to find solution
+    operation GroversAlgorithm (V : Int, K : Int, maxIterations : Int, 
+        oracle : ((Qubit[], Qubit) => Unit is Adj)) : Int[] {
+        // This task is similar to task 2.2 from SolveSATWithGrover kata, 
+        // but the percentage of correct solutions is potentially higher.
+        mutable coloring = new Int[V];
+
+        // Note that coloring register has the number of qubits that is 
+        // twice the number of vertices (K qubits per vertex).
+        using ((register, output) = (Qubit[K * V], Qubit())) {
+            mutable correct = false;
+            mutable iter = 1;
+            repeat {
+                Message($"Trying search with {iter} iterations");
+                GroversAlgorithm_Loop(register, oracle, iter);
+                let res = MultiM(register);
+                // to check whether the result is correct, apply the oracle to the 
+                // register plus ancilla after measurement
+                oracle(register, output);
+                if (MResetZ(output) == One) {
+                    set correct = true;
+                    // Read off coloring
+                    set coloring = MeasureColoring(K, register);
+                }
+                ResetAll(register);
+            } until (correct or iter > maxIterations)  
+            fixup {
+                set iter += 1;
+            }
+            if (not correct) {
+                fail "Failed to find a coloring";
+            }
+        }
+        return coloring;
+    }
+
+    /// # Summary
+    /// Grover loop implementation taken from SolveSATWithGrover kata.
```

Comment:
You can link the Kata here: https://github.com/microsoft/QuantumKatas/tree/master/SolveSATWithGrover
Also here I think adding the input and output variable descriptions into the docblock would be helpful, as it gets picked up by our API documentation.